src/hooks/useNavigation.ts

```diff
@@ -0,0 +1,31 @@
+import { useMemo } from 'react'
+import createNavigationHelpers from '../helpers/createNavigationHelpers'
+import useComponentId from './useComponentId'
+
+/**
+ * Returns a set of action helpers for
+ * [component](https://wix.github.io/react-native-navigation/api/component),
+ * [root](https://wix.github.io/react-native-navigation/api/root),
+ * [stack](https://wix.github.io/react-native-navigation/api/stack),
+ * [modal](https://wix.github.io/react-native-navigation/api/modal),
+ * [overlay](https://wix.github.io/react-native-navigation/api/overlay)
+ * Navigation API
+ */
+const useNavigation = (
+  /**
+   * Component reference id. Used to give context to the Navigation functions that requires componentId parameter.
+   */
+  componentId?: string
+) => {
+  const componentIdFromProvider = useComponentId()
+
+  const id = componentId || componentIdFromProvider
+
+  if (!id) {
+    throw new Error('Missing componentId. Use ComponentIdContext or pass componentId as argument.')
```

Comment:
I'm wondering if it's ok to throw an error from a hook.
Could happen the `componentId` didn't come on the first render?
Currently it's there on the first render but I don't know if that's guaranteed.

Reply:
I would assume `componentId` will never be empty since:

1. `componentId` is provided through the `withNavigationContext` / `NavigationProvider` when registering RNN screens
2. `componentId` is provided manually on the screen using the injected `componentId` prop that will never be empty as it's being injected by RNN on screen registration

I wouldn't expect someone changing the value of the `componentId` manually  

Thoughts?